api/api.py

```diff
@@ -1,43 +1,105 @@
+from api.helpers import get_token
 
 api = Blueprint('api', 'api', url_prefix='/api/v1/')
 store = store()
-@api.route('auth/register',methods = ['POST'])
+
+
+def auth(n):
+    """ Auth middleware to check logged in user"""
+    @wraps(n)
+    def wrap(*args, **kwargs):
+        if request.headers.get('Authorization'):
+            token = request.headers.get('Authorization')
+            if user().token_exists(token):
+                return n(*args, **kwargs)
+        response = jsonify({
+            'status': 'error',
+            'message': "Unauthorized"
+        })
+        response.status_code = 401
+        return response
+    return wrap
+
+
+@api.route('auth/register', methods=['POST'])
 def register():
-    inputs = RegisterInputs()
+    """
+        Registration endpoint method
+    """
+    inputs = register_inputs()
     if not inputs.validate(request.form):
         response = jsonify(status='error', message=inputs.error)
         response.status_code = 400
         return response
     data = {
-        'id':uuid.uuid4().hex,
-        'names':request.form['names'],
-        'email':request.form['email'],
-        'password':generate_password_hash(request.form['password']),
+        'id': uuid.uuid4().hex,
+        'names': request.form['names'],
+        'email': request.form['email'],
+        'password': request.form['password'],
     }
-    if (user().user_exists(data['email'])):
+    if user().user_exists(data['email']):
         response = jsonify({
-            'status':'error',
-            'message':"Sorry the email address has been taken"
+            'status': 'error',
+            'message': "Sorry the email address has been taken"
         })
         response.status_code = 400
         return response
     user().save(data)
     response = jsonify({
-        'status':'ok',
-        'message':"You have been successfully registered"
+        'status': 'ok',
+        'message': "You have been successfully registered"
     })
     response.status_code = 200
     return response
-"""
-    login Endpoints
-"""
-@api.route('auth/login',methods = ['POST'])
-# @auth
+
+
+@api.route('auth/login', methods=['POST'])
 def login():
-    pass
\ No newline at end of file
+    """
+        Login endpoint method
+    """
+    inputs = login_inputs()
+    if not inputs.validate(request.form):
+        response = jsonify(status='error', message=inputs.error)
+        response.status_code = 400
+        return response
+    data = {
+        'email': request.form['email'],
+        'password': request.form['password'],
+    }
+    # Check if email exists in the store
+    logged_user = user().get_user(data['email'])
+    if logged_user:
+        # Check password
+        if check_password_hash(logged_user['password'], data['password']):
+            token = get_token(logged_user['id'])
+            user().add_token(token)
+            response = jsonify({
+                'status': 'ok',
+                'token': token,
```

Comment:
Is the token being sent with the response data or is it set in the response headers?

Reply:
Fixed